src/components/Body/Body.js

```diff
@@ -0,0 +1,76 @@
+import React, { useState, useEffect } from 'react';
+import Trials from './Trials/Trials';
+import Results from './Results/Results';
+import classes from './Body.module.css';
+
+const GAME_STATUS = {
+    NOT_STARTED: 0,
+    ACTIVE: 1,
+    WON: 2,
+    LOST: 3
+}
+
+const Body = () => {
+
+    //STATES & HOOKS-------------------------------------------------------------------
+    const [secretCode, setSecretCode] = useState("0000");
+    const [selectedIndex, setSelectedIndex] = useState(0); //this state holds the current enabled row  
+    const [gameStatus, setGameStatus] = useState(GAME_STATUS.NOT_STARTED);
+    const [resultText, setResultText] = useState(""); //the result text that indicates if the user won or lost the game
+    const [clearScreen, setClearScreen] = useState(true); //to reset the screen when the user starts the game again
+
+
+    useEffect(() => { //control the status of the game
+        switch (gameStatus) {
+            case GAME_STATUS.ACTIVE:    //this restarts the game and clears the screen
+                setResultText("");
+                setSecretCode(generateRandomNumber(1000, 9999)); //set a new random secret code
+                setSelectedIndex(1); //to enable the first row again
+                setClearScreen(true);
+                return;
+
+            case GAME_STATUS.WON:
+                setResultText("Congrats, You won the game!");
+                return;
+
+            case GAME_STATUS.LOST:
+                setResultText("Game Over, You lost the game :(");
+                return;
+
+            default:
+                return;
+        }
+    }, [gameStatus]);
+
+
+    //FUNCTIONS-----------------------------------------------------------------
+    const generateRandomNumber = (min, max) => { //generate number between min (inclusive) and max (inclusive)
+        return Math.floor(Math.random() * (max - min + 1)) + min;
+    };
+
+    const startButtonClickHandler = () => { //to start the game
+        setGameStatus(GAME_STATUS.ACTIVE);
+    };
+
+
+    //JSX CODE-----------------------------------------------------------------
+    return (
+        <div className={classes.mainContainer}>
+
+            {/* RENDER THE 8 ROWS*/}
+            <Trials selectedIndex={selectedIndex} setSelectedIndex={setSelectedIndex}
+                secretCode={secretCode}
+                setGameStatus={setGameStatus}
+                clearScreen={clearScreen} setClearScreen={setClearScreen}
+                GAME_STATUS_WON={GAME_STATUS.WON} GAME_STATUS_LOST={GAME_STATUS.LOST} />
```

Comment:
No, since the GAME_STATUS is a shared const between several components, create new file under body folder called "body.service.js" and move the const to it and then use it in any component you need

Reply:
done